src/rgw/rgw_op.cc

```diff
@@ -1034,11 +1034,15 @@ int handle_cloudtier_obj(req_state* s, const DoutPrefixProvider *dpp, rgw::sal::
       // fill in the entry. XXX: Maybe we can avoid it by passing only necessary params
       rgw_bucket_dir_entry ent;
       ent.key.name = s->object->get_key().name;
+      ent.key.instance = s->object->get_key().instance;
       ent.meta.accounted_size = ent.meta.size = s->obj_size;
       ent.meta.etag = "" ;
       ceph::real_time mtime = s->object->get_mtime();
```

Comment:
Not whether we need this anymore, mtime is again filled in the restore_obj_from_cloud in rados and rados_sal layer

Reply:
yes thanks! I cleaned up the API.